plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -124,7 +125,8 @@ private static boolean isTrackingIdDefined() {
         && Constants.ANALYTICS_TRACKING_ID.startsWith("UA-");
   }
 
-  private String getAnonymizedClientId() {
+  @VisibleForTesting
+  String getAnonymizedClientId() {
     String clientId = preferences.get(AnalyticsPreferences.ANALYTICS_CLIENT_ID, null);
     if (clientId == null) {
       clientId = UUID.randomUUID().toString();
```

Comment:
Or since the preferences comes from ConfigurationScope (shared across all instances of this product), two Eclipse workbenches.  (I often have 2-3 workbenches at once.)  Could compute this in the constructor?